### ISSUE TITLE
Port r2-retdec to cutter

### ISSUE THREAD

**Issue by Maijin** [radareorg/cutter#203]
https://github.com/jpenalbae/r2-scripts/blob/master/decompiler/decompile.js as another pseudo-code panel capability in prevision of the full release of retdec.

see https://asciinema.org/a/20904

![image](https://user-images.githubusercontent.com/1408600/33819640-c14aa14e-de4b-11e7-928a-270a3a7cc826.png)

**Comment by thestr4ng3r** [comment]
Not sure if it makes sense to integrate that as a core feature since it sends your stuff to that api. I think it would make more sense as an optional plugin of some sort.

Adding a plugin interface to Cutter should not be too hard: http://doc.qt.io/qt-5/plugins-howto.html#the-low-level-api-extending-qt-applications
However, the interfaces should to be well thought through, so they can be kept stable.

**Comment by Maijin** [comment]
yep that's what i was thinking too and i think retdec could be a nice first "plugin"

**Comment by XVilka** [comment]
Or may be just wait until they release promised sources?

**Comment by jpenalbae** [comment]
I agree that making r2pipe plugins as core features does not really make sense. Instead cutter should provide a way to execute plugins within the UI, some possible features could be:
* Recent scripts window, to be able to rerun latest used scripts.
* Some way to provide arguments for the scripts
* Option to associate scripts with keybindings, and with aliases such as `#!pipe script.js -arg1 -arg2 @ $$`

**Comment by Maijin** [comment]
@XVilka the interface still need to be designed and will be same, API or not.
@jpenalbae Well @thestr4ng3r proposed to make a new plugin system for cutter which makes sense as r2pipe plugin won't contain the UI stuff

**Comment by Maijin** [comment]
Hey that could easily be solved by using Jupyter actually #39  !

**Comment by jpenalbae** [comment]
IMHO I would first support normal r2pipe to be able to work with existing scripts, and then create a new plugins system with UI support. Being able to use normal r2pipe scripts is a must.

**Comment by radare** [comment]
Agree. Adding a scripting panel wih a text area and a play button is easy and necessary

**Comment by Maijin** [comment]
But should be done with jupyter too.

**Comment by Maijin** [comment]
To get such stuff http://ipywidgets.readthedocs.io/en/stable/examples/Widget%20List.html

**Comment by stonedreamforest** [comment]
refer：https://github.com/avast-tl/retdec
